The-SMIRNOFF-force-field-format.md

```diff
@@ -51,26 +75,85 @@ If not specified by a `*_unit` attribute, the [standard OpenMM unit system](http
+
+For example, to assign partial charges
+```
+<LibraryCharges charge_unit="elementary_charge">
+   <!-- match a non-terminal alanine residue -->
+   <Residue name="ALA" smirks="[$([NX3H:1](C)(C)[H:2])][CX4H:3]([CH3X4:4]([H:5][H:6][H:7]))[CX3:8](=[OX1:9])([N])" charge1="-0.4157" charge2="0.2719" charge3="0.0337" charge4="-0.1825" charge5="0.0603" charge6="0.0603" charge7="0.0603" charge8="0.5973" charge9="-0.5679">
+   ...
+</LibraryCharges>
+```
+In this case, a SMIRKS string defining the residue tags each atom that should receive a partial charge, with the charges specified by `charge#` attributes.
+Note that the first match for a given set of atoms is used, so chemically equivalent atoms should be assigned the same charge to avoid undefined behavior.
+
+Solvent models or excipients can also have partial charges specified in this manner:
+```
+<LibraryCharges charge_unit="elementary_charge">
+   <!-- TIP3P water oxygen with charge override -->
+   <!-- WARNING: charges are not supported yet (see issue openforcefield#25), and they are currently here only for documentation -->
+   <Atom smirks="[#1:1]-[#8X2H2+0:2]-[#1:3]" charge1="+0.417" charge2="-0.834" charge3="+0.417"/>
+</LibraryCharges>
+```
+
+**QUESTION:** Is this an OK way to handle chemically equivalent atoms, like hydrogens? Is there a simpler way to avoid having to specify multiple copies of chemically equivalent atoms?
+
+### Small molecule charges
+
+In keeping with the AMBER force field philosophy, especially as implemented in small molecule force fields such as [GAFF](http://ambermd.org/antechamber/gaff.html), [GAFF2](https://mulan.swmed.edu/group/gaff.php), and [parm@Frosst](http://www.ccl.net/cca/data/parm_at_Frosst/), partial charges for small molecules are assigned using a semiempirical method (e.g. AM1) coupled to a population analysis scheme (e.g. CM2) and corrected via a bond charge correction step, as in the highly successful [AM1-BCC](https://dx.doi.org/10.1002/jcc.10128) approach.
+
+Future iterations of SMIRNOFF will provide support for multiconformer RESP fitting to high-level QM.
+
+### Automated generation of biopolymer residue charges
+
+Future iterations of SMIRNOFF will provide various schemes for automatically fragmenting and charging residues of biopolymers or very large small molecules.
+
+### Prespecified charges (reference implementation only)
+
+In our reference implementation of SMIRNOFF in the `openforcefield` toolkit, we also provide a method for specifying partial charges within the `Molecule` objects in the `Topology`.
+This method is provided solely for convenience in exploring alternative charging schemes; actual forcefield releases for distribution will use one of the other mechanisms specified above.
+
+**QUESTION:** Do we still want to support being able to pull charges from `OEMol`s for convenience, or is the `<LibraryCharges>` mechanism sufficiently general?
```

Comment:
We DEFINITELY need to allow people to use charges from provided molecules (OEMol, RDMol, whatever) for the purposes of force field science. Otherwise to try any new charge sets, we and/or others will need to implement them into SMIRNOFF or encode them as library charges before we can try them, which would be a major bottleneck and impede testing and science.  (e.g. people exploring new charge models may want to easily use them with our FF, as they would with GAFF, and it's easiest if they can just load them from molecules...)

Reply:
Yes we must allow people to use charges from provided molecules.